spatialmath/dualquaternion.go

```diff
@@ -185,6 +186,39 @@ func (q *dualQuaternion) Transformation(by dualquat.Number) dualquat.Number {
 	}
 }
 
+// Format implements fmt.Formatter to allow for finer grain control over printing.
+// dualquat.Number also has this implemented so in order to get custom printing we need to also implement this as opposed to simply
+// implementing fmt.Stringer.
+// Some example verbs and how they would display a zero pose are shown below
+//
+//	Verb    | String
+//	%v		| {X:0 Y:0 Z:0 OX:0 OY:0 OZ:1 Theta:0°}
+//	%.3v	| {X:0.000 Y:0.000 Z:0.000 OX:0.000 OY:0.000 OZ:1.000 Theta:0.000°}
+//	%#v		| r3.Vector{0 0 0} *spatialmath.OrientationVectorDegrees{0 0 0 1}
+//	%g		| ((1+0i+0j+0k)+(+0+0i+0j+0k)ϵ)
+func (q *dualQuaternion) Format(fs fmt.State, verb rune) {
+	prec, _ := fs.Precision()
+	width, _ := fs.Width()
+	format := fmt.Sprintf("%%%d.%df", width, prec)
+	pt := q.Point()
+	o := q.Orientation().OrientationVectorDegrees()
+	switch verb {
+	case 'v':
+		if fs.Flag('#') {
+			//nolint:errcheck
+			fmt.Fprintf(fs, "%T"+format+" %T"+format, pt, pt, o, *o)
+			return
+		}
+		fallthrough
+	case 's':
+		format = fmt.Sprintf("{X:%s Y:%s Z:%s OX:%s OY:%s OZ:%s Theta:%s°}", format, format, format, format, format, format, format)
```

Comment:
Could we document this above the function as well? This replicates what PoseToProtobuf looks like, right?

Reply:
Yeah its pretty much the same thing as calling PoseToProtobuf, I will add to the function description

Reply:
But with a couple bells and whistles, like variable precision printing and also a degree symbol for ease of knowing what representation the OV is in